onadata/libs/serializers/data_serializer.py

```diff
@@ -303,3 +303,48 @@ def create(self, validated_data):
                                      validated_data['id_string'])
 
         return instance
+
+
+class FLOIPSubmissionSerializer(SubmissionSuccessMixin,
+                                serializers.Serializer):
+    """
+    FLOIP SubmmissionSerializer - Handles a row of FLOIP specification format.
+    """
+    def validate(self, attrs):
+        data = self.context['request'].data
+        error_msg = None
+
+        if not isinstance(data, list):
+            error_msg = u'Invalid format. Expecting a list.'
+        else:
+            for row_i in xrange(len(data)):
+                if len(data[row_i]) != 6:
+                    error_msg = u"All rows must have 6 values, "\
+                                u"row %s does not." % row_i
+                    break
+
+        if error_msg:
+            raise serializers.ValidationError(_(error_msg))
+
+        return super(FLOIPSubmissionSerializer, self).validate(attrs)
+
+    def to_internal_value(self, data):
+        """
+        Overrides validating rows in list data.
+        """
+        data = {data[1]: data}
```

Comment:
I feel like we should have a check first that data has at least one row and is probably not `None`.

Reply:
By this point, we get a row of data already. I can add the check at this point to confirm that.

```
elif len(data) > 0:
    for row_i in xrange(len(data)):
        if len(data[row_i]) != 6:
            error_msg = _(u"All rows must have 6 values, "\
                          u"row %(row)s does not." % {'row': row_i})
```

Reply:
More likely,
```python
elif data:
    for row_i, row in enumerate(data):
        if len(row) != 6:
            error_msg = _(u"All rows must have 6 values, "\
                          u"row %(row)s does not." % {'row': row_i})
```